### PR TITLE
Add all and any in DataFrame

### PR DESCRIPTION
**Pull request by HyukjinKwon**
This PR adds all and any in DataFrame

Also, this PR fixes `Series.quantile` working with Spark 2.3 by avoiding to use `arrays_zip` expression.

### CODE REVIEW

File: databricks/koalas/frame.py [hunk 1]
```diff
@@ -5510,6 +5510,164 @@ def melt(self, id_vars=None, value_vars=None, var_name='variable',
...
 
+    # TODO: axis, skipna, and many arguments should be implemented.
+    def all(self, axis: Union[int, str] = 0) -> bool:
+        """
+        Return whether all elements are True.
+
+        Returns True unless there at least one element within a series that is
+        False or equivalent (e.g. zero or empty)
+
+        .. note:: this API requires Spark 2.4+.
```
Comment (HyukjinKwon) [inline databricks/koalas/frame.py]: Let me try to work around this ..

File: databricks/koalas/frame.py [hunk 2]
```diff
@@ -5510,6 +5510,166 @@ def melt(self, id_vars=None, value_vars=None, var_name='variable',
 
         return DataFrame(exploded_df)
 
+    # TODO: axis, skipna, and many arguments should be implemented.
+    def all(self, axis: Union[int, str] = 0) -> bool:
+        """
+        Return whether all elements are True.
+
+        Returns True unless there at least one element within a series that is
```
Comment (ueshin) [inline databricks/koalas/frame.py]: nit: `there is at least`?

File: databricks/koalas/frame.py [hunk 3]
```diff
@@ -5510,6 +5510,166 @@ def melt(self, id_vars=None, value_vars=None, var_name='variable',
...
+
+        # TODO: there is a similar logic to transpose in, for instance,
+        #  DataFrame.any, Series.quantile. Maybe we should deduplicate it.
+        sdf = self._sdf
+        internal_index_column = "__index_level_0__"
+        value_column = "value"
+        cols = []
+        for data_column, applied_col in zip(data_columns, applied):
+            cols.append(F.struct(
+                F.lit("%s" % data_column).alias(internal_index_column),
```
Comment (ueshin) [inline databricks/koalas/frame.py]: nit: do we need to format `data_column`?

**Comment by ueshin** [comment]
Thanks! merging.

**Pull request merged**